homeassistant/components/incomfort/config_flow.py

```diff
@@ -102,6 +107,33 @@ def async_get_options_flow(
         """Get the options flow for this handler."""
         return InComfortOptionsFlowHandler()
 
+    async def async_step_dhcp(
+        self, discovery_info: DhcpServiceInfo
+    ) -> ConfigFlowResult:
+        """Prepare configuration for a DHCP discovered Intergas Gateway device."""
+        self._discovered_host = discovery_info.ip
+        self._async_abort_entries_match({CONF_HOST: self._discovered_host})
```

Comment:
We can also check if we have a host for this loaded, and then we can add the Mac address to the entry data, which we then can use for the device info

Reply:
Existing entries have no mac address registered, and the serial number is not that of the gateway. So I think this double check should work.

Reply:
Existing entries with a host match are migrated now and will be assigned a unique id based on the gateway mac. That allows to auto update the IP in case DHCP offers a new IP address.